### DISCUSSION TITLE
请问如何在router.beforeeach里面使用message？

### DISCUSSION THREAD

**Discussion by mollerzhu** [tusen-ai/naive-ui#3186] [General]
我看到了组件说明里的QA，说挂在到全局window.$message里，但是还是不知道如何在router里面使用，感谢帮助~

**Comment by 07akioni** [comment]
如果自己想不到处理方式的话可以用这个：

https://www.naiveui.com/zh-CN/os-theme/components/discrete

**Reply by 07akioni** [reply]
处理起来更简单

**Reply by mollerzhu** [reply]
已解决，感谢
